### DISCUSSION TITLE
API endpoint not being hit on Vercel

### DISCUSSION THREAD

**Discussion by betaupsx86** [vercel/community#7271] [Help] [answered]
### Summary

Hi,
I deployed an app to vercel and my API serverless function is not being hit. The endpoint is a webhook I'm using for Stripe. Locally it works perfectly ok. I've tried to reach it manually as well. Nothing works and I don't see the logs and return status that I've set up. The only log I see is when I go to the address in my browser and I get WARNING GET method invalid, which makes sense since it is a POST function.
Any help is appreciated

Not sure if this is helpful. But it shows 0kb in the build logs? All other pages static or dynamic work well
![image](https://github.com/vercel/community/assets/22305406/dd8da42f-c353-41ec-ac28-398d3f7488af)

### EDIT
When I use something like reqbin.com to POST I see a html page with Vercel Authenticating spinner. Am I setting the endpoint correctly? Again, this works perfectly fine locally with **npm run dev**
My endpoint is in the app dir **app/api/webhooks/route.ts**. Just wondering if that has something to do with it. First time using the platform

### Steps to Reproduce

Not sure how to reproduce other than deploying and sending POST requests with curl or other online tools

**Comment by versecafe** [comment]
It should say 0kb as the size is the total client side JS bundle that gets sent which for API routes should always be 0, can you share your production deployment URL & git repo?

**Reply by betaupsx86** [reply]
Hey. This got solved after promoting the preview deployment to production.
Do API routes not work in preview?

**Reply by versecafe** [reply, marked as answer]
They work in preview it's just they hit auth walls unless they have a special bypass token, so external webhooks unless you add that token won't work in preview deploys unless you make all previews public

**Reply by betaupsx86** [reply]
I see. that makes sense and explains the auth spinner I was seeing. How do I add that token? or is the only way to make the preview public?

**Reply by amyegan** [reply]
You would use the [system environment variable](https://vercel.com/docs/projects/environment-variables/system-environment-variables#:~:text=VERCEL_AUTOMATION_BYPASS_SECRET) `VERCEL_AUTOMATION_BYPASS_SECRET` in a query parameter or HTTP header named `x-vercel-protection-bypass`.

You can find more info about Protection Bypass for Automation here: https://vercel.com/docs/security/deployment-protection/methods-to-bypass-deployment-protection/protection-bypass-automation

**Reply by betaupsx86** [reply]
Thanks, trying this now

[a comment by an automated account is not shown]
